src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -382,6 +381,26 @@ public bool ProcessTransaction(Transaction transaction, int? blockHeight = null,
                     }
                 }
 
+                // Check if we're trying to spend a utxo twice
+                foreach (TxIn input in transaction.Inputs)
+                {
+                    if (!this.outpointLookup.TryGetValue(input.PrevOut, out TransactionData tTx))
+                    {
+                        continue;
+                    }
+
+                    // If we're trying to spend an input that is already spent, and it's not coming in a new block, don't reserve the transaction. 
+                    // This would be the case when blocks are synced in between CrossChainTransferStore calling
+                    // FederationWalletTransactionHandler.BuildTransaction and FederationWalletManager.ProcessTransaction.
+
+                    // TODO: Do we need to get the transaction from the wallet rather than using tTx? Copying the other method approaches.
```

Comment:
It should be the same reference to the underlying `TransactionData` instance, so it shouldn't make a difference in my opinion.